arch/x86/core/ia32/swap.S

```diff
@@ -66,9 +66,9 @@
  */
 
 SECTION_FUNC(TEXT, arch_swap)
-#if defined(CONFIG_TRACING)
+#if defined(CONFIG_INSTRUMENT_THREAD_SWITCHING)
 	pushl	%eax
```

Comment:
not strictly part of this PR. but are we using EAX for something here? IA32 passes in all args via stack so maybe this is unnecessary